app/services/calories/fetch_product_calories.rb

```diff
@@ -0,0 +1,29 @@
+module Calories
+
+  class FetchProductCalories
+
+    def initialize(product)
```

Comment:
please add api_client as argument in constructor:

```
def initialize(product, api_client)
  @api_client = api_client
end
```

and later use it as usual in `fetch_product_calories` method

```
api_client.get_product_data(@product.name)
```

it's just refactoring the code to follow dependency injection pattern.

you will also need to change where it's called in product model to pass argument

```
  def fetch_calories
    client = Nutritionix::ApiClient.new
    Calories::FetchProductCalories.new(self, client).call
  end
```